api/client/src/main/java/org/projectnessie/client/auth/oauth2/TokensResponseBase.java

```diff
@@ -33,31 +33,43 @@
  *
  * <p>It is also a flattened representation of a {@link Tokens} pair.
  */
-interface TokensResponseBase extends Tokens {
+interface TokensResponseBase {
 
-  @Value.Lazy
-  @JsonIgnore
-  @Override
-  default AccessToken getAccessToken() {
-    return ImmutableAccessToken.builder()
-        .tokenType(getTokenType())
-        .payload(getAccessTokenPayload())
-        .expirationTime(getAccessTokenExpirationTime())
-        .build();
-  }
+  /** Convert this response to a {@link Tokens} pair using the provided clock. */
+  default Tokens asTokens(Supplier<Instant> clock) {
 
-  @Nullable
-  @Value.Lazy
-  @JsonIgnore
-  @Override
-  default RefreshToken getRefreshToken() {
-    if (getRefreshTokenPayload() != null) {
-      return ImmutableRefreshToken.builder()
-          .payload(getRefreshTokenPayload())
-          .expirationTime(getRefreshTokenExpirationTime())
-          .build();
-    }
-    return null;
+    Duration accessExpiresIn = getAccessTokenExpiresIn();
+    AccessToken accessToken =
+        ImmutableAccessToken.builder()
+            .tokenType(getTokenType())
+            .payload(getAccessTokenPayload())
+            .expirationTime(accessExpiresIn == null ? null : clock.get().plus(accessExpiresIn))
```

Comment:
Nit: Add an external variable `Instant now = clock.get()` at the beginning of `asTokens()` and use it here and below